### PR TITLE
feat: feedback widget from developers modal

### PR DESCRIPTION
**Pull request by pdp2121**
Add back feedback widget and fix design:

Dark Mode:

Screenshot 2023-04-26 at 2 16 12 PM

Screenshot 2023-04-26 at 3 35 14 PM

Light Mode:

Screenshot 2023-04-26 at 2 13 55 PM

Screenshot 2023-04-26 at 3 34 50 PM

### CODE REVIEW

**Comment by pdp2121** [comment]
Hmm the Preview didn't reflect the changes here

**Comment by pdp2121** [comment]
@mDuo13 The preview doesn't work because we are using a vendor js script `feedback-widget.js` that couldn't seem to load properly. Did we have this issue before the widget was commented out?

File: template/component-feedback-widget.html.jinja [hunk 1]
```diff
@@ -1,19 +1,19 @@
 <!-- Feedback Widget -->
-<!-- <script src="{{currentpage.prefix}}assets/vendor/feedback-widget.js"></script> -->
-<!-- <div id="feedback-content"></div> -->
+<script src="../../assets/vendor/feedback-widget.js"></script>
```
Comment (mDuo13) [inline template/component-feedback-widget.html.jinja]: This change (replacing `{{currentpage.prefix}}` with `../../`) is why the feedback widget script is not loading as expected. If you revert this part, it should be fine.

**Review by mDuo13 — CHANGES_REQUESTED** [review]

**Comment by pdp2121** [comment]
@mDuo13 the preview is ready for reviews. You can check this link:
https://xrplf.github.io/xrpl-dev-portal/pr-preview/feedback-widget/xrpl-quickstart.html

**Comment by mDuo13** [comment]
The thumbs-up/down icons aren't showing up in the preview, but they should work in prod (absolute URLs will only work in prod, relative URLs would work in both prod and staging—I think these URLs are coming from the JS for the feedback widget somehow). So could be improved but it's OK as-is.

Placement of the popup on mobile is a bit off, and in general having it as a popup is kind of annoying as it obstructs important parts of the page before you've even had a chance to read them and decide if the page was helpful.

I seem to recall that at some point we had the widget "docked" to below the main article in the center column so it wasn't a popup at all, but something seems to have changed to make that not apply any more. One way to do that again is to add these rules to the CSS:

```css
#feedback-content .docked-widget {
    position: static !important;
    box-shadow: none !important;
    width: auto !important;
}
```

Also, in light mode, the "How can we do better?" text area has white text on a white background. You can fix that with:

```scss
html.light #feedback-content .widget-form-wrapper textarea {
    color: $black;
}
```

I think with those adjustments, this should be almost good to go—one other thing to look out for is conflicts / versioning of the CSS file. [This comment](https://github.com/XRPLF/xrpl-dev-portal/pull/1866#discussion_r1191611959) explains general steps for incrementing the CSS file, but because both this PR and #1866 need to modify the CSS file, one of them needs to rebase onto the other and rebuild with the SCSS edits from both. (If we deploy the edits together, then only one version increment is needed; otherwise, we need to increment twice.)

**Comment by pdp2121** [comment]
@mDuo13 Thanks for your feedbacks! I'm not sure why my changes in this commit to make the widget "docked" below center section was not reflected on the preview (rules did not show up on Inspect as well):
https://github.com/XRPLF/xrpl-dev-portal/pull/1892/commits/937e4caf35a94181b5ab6faef6c2431e601cfe4e
even though it worked locally.

Also, the "How can we do better?" text area is black in the preview on my end, with the following code:
`.widget-form-wrapper {
    textarea {
      background-color: $gray-100 !important;
    }

    .widget-header-title {
      color: $gray-900 !important;
    }
}`
in `_light-theme.scss`

I will coordinate with Caleb and Jackson on CSS file versioning.

**Comment by pdp2121** [comment]
Preview was fixed. widget should be docked now.

**Comment by ckniffen** [comment]
The text color is white so I cant see what I typed out.  The dots are spellcheck.
Monosnap Tokens - XRPL org 2023-05-18 13-12-51

**Comment by pdp2121** [comment]
> The text color is white so I cant see what I typed out. The dots are spellcheck. Monosnap Tokens - XRPL org 2023-05-18 13-12-51

@ckniffen Can you try again? It worked for me as I tested just now in preview. It could be preview issue.

Screenshot 2023-05-18 at 6 09 24 PM

**Comment by pdp2121** [comment]
@mDuo13 I think you are right. Caleb and I just figured that the color of the textarea changes based on OS theme color so we should have a fixed color for it. I will make the change you (and Caleb) suggested now.

File: styles/light/_light-theme.scss [hunk 2]
```diff
@@ -763,15 +763,46 @@ pre code {
 
 // feedback widget
 #feedback-content {
-  .docked-widget {
+  // .docked-widget {
```
Comment (ckniffen) [inline styles/light/_light-theme.scss]: delete commented out code.

**Review by ckniffen — APPROVED** [review]

**Comment by intelliot** [comment]
Where does the feedback go, and who will review it?

**Comment by pdp2121** [comment]
> Where does the feedback go, and who will review it?

@intelliot It will be sent to Google Analytics (I'm not sure who's going to review though). The previous portal that we used for feedbacks is deprecating next month. There's has been a GA4 version set up for xrpl.org, and we will need to add mappings for feedbacks. I'm reaching out to people to handle this.

**Comment by mDuo13** [comment]
The preview looks good, but can you rebase on the latest master and rebuild the CSS to resolve merge conflicts?

**Review by mDuo13 — APPROVED** [review]

**Comment by pdp2121** [comment]
This can be merged now as we have successfully created data layer push event for gtm, as discussed with Caleb and Atilla.

**Pull request merged**